ffi/firewood.go

```diff
@@ -4,7 +4,15 @@
 package firewood
 
 // // Note that -lm is required on Linux but not on Mac.
-// #cgo LDFLAGS: -L${SRCDIR}/../target/release -L/usr/local/lib -lfirewood_ffi -lm
+// #cgo linux,amd64 LDFLAGS: -L${SRCDIR}/libs/x86_64-unknown-linux-gnu -lm
+// #cgo linux,arm64 LDFLAGS: -L${SRCDIR}/libs/aarch64-unknown-linux-gnu -lm
+// #cgo darwin,amd64 LDFLAGS: -L${SRCDIR}/libs/x86_64-apple-darwin
+// #cgo darwin,arm64 LDFLAGS: -L${SRCDIR}/libs/aarch64-apple-darwin
+// // XXX: last search path takes precedence, which means we prioritize
+// // local builds over pre-built and maxperf over release build
+// #cgo LDFLAGS: -L${SRCDIR}/../target/release
```

Comment:
This is a great idea, we should also include debug in this list. That is, if you have a debug build, it will use that. If not, it will use the release build, and if not, the maxperf build. This would avoid some problems I ran into when I did a "cargo build" and forgot to ask for a release build before retesting ffi.

We should make this very clear in the development workflows.

Reply:
Also as mentioned in the comment - the last search path takes precedence, so the ordering line by line of:

1. Release
2. Maxperf

Means that maxperf takes precedence over release as you requested previously. Happy to modify the ordering if you prefer.

For now, I am adding `debug` to the top of that list going with my understanding of the previous motivation to prioritize using `maxperf` whenever possible.

Maybe you meant that only for the static builds we are pushing for the combination of smaller binaries and performance, but prefer debug to be prioritized when testing locally?

Reply:
Yes, and just to avoid the stupid mistake of `cargo build && go test .` that I have made several times. I think you also get better stack traces when you're debugging with debug builds.